### ISSUE TITLE
Build errors when compiling with Free Pascal

### ISSUE THREAD

**Issue by magorium** [todaysoftware/libgit2-delphi#3]
Hi there,

Thank you very much for your up-to-date bindings.

I ran into a couple of issues when compiling with FPC 3.2.3. Tested on Linux x86_64. 

The **first issue** is concerning the uses clause of unit LibGit2.
```
uses
  {$IF RTLVersion >= 23}
  System.SysUtils;
  {$ELSE}
  SysUtils;
  {$IFEND}
```

The if condition does not compile due to the fact of missing RTLVersion define (FPC will choke on that) so I needed to change that part to:
```
uses
  {$IFDEF DELPHI}
    {$IF RTLVersion >= 23}
  System.SysUtils;
    {$ENDIF}
  {$ELSE}
  SysUtils;
  {$ENDIF}
```

Please note that in (near) future FPC will also support dotted RTL unit names (so the unit names are then similar as for Delphi). There is a FPC define FPC_DOTTEDUNITS that can be used to determine if dotted RTL is active or not. see also https://wiki.freepascal.org/FPC_Dotted_RTL

The **second issue** that prevents the project to compile out of the box is that also FPC does not have size_t declared. It is a platform specific type (for example unix/linux base units have it declared) but then you would have to include those platform specific units (which kinda defeats the purpose).

so I needed to remove the optional condition
```
  {$IFNDEF FPC}
type
  size_t = uintptr_t;
  {$ENDIF}
```
so that the code looks like
```
type
  size_t = uintptr_t;
```

Would be nice if these issues could be addressed.

TIA

**Comment by todaysoftware** [comment]
Thanks for the info, I do have free pascal on my laptop so should be able to fix up these things all OK and confirm it will compile.

**Comment by todaysoftware** [comment]
I realised I had compiled it from the Lazarus IDE previously and it all compiled fine. However just using the FPC compiler I got the RTLVersion error, however I never got the size_t error which was interesting. Anyway made some changes that should cope with both issues and confirmed it is compiling under both Delphi and FPC.

3a84a46

**Comment by magorium** [comment]
Confirmed that your changes are working. 

I have (also) no idea what's up with the size_t declaration. I tried to pinpoint what might have changed but was unable to locate any (recent) clues.

Your changes are more elegant than I could have imagined myself and nicely takes care of all situations (worthy  of another star :+1:  ).

I am closing this issue (open it at your convenience) and thank you very much for your time.

regards.
